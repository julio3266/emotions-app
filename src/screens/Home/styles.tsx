import styled from 'styled-components/native';

export const Container = styled.View`
  flex: 1;
  background-color: ${({theme}) => theme.colors.blueViolet};
`;

export const Content = styled.View`
  top: 4.5%;
  flex: 1;
  display: flex;
  align-items: center;
  justify-content: flex-start;
`;

export const Row = styled.View`
  flex-direction: row;
`;

export const Column = styled.View`
  flex-direction: column;
`;
