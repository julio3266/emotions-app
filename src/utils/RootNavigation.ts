import {createRef} from 'react';

export const navigationRef: any = createRef();

export const navigate = (name: string, params?: any) => {
  navigationRef.current?.navigate(name, params);
};

export const goBack = () => {
  navigationRef.current?.goBack();
};

export const params = () => {
  navigationRef.current?.params;
};
