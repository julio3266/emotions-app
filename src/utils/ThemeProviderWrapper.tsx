import React from 'react';
import {ThemeProvider} from 'styled-components/native';
import Theme from '../theme/styles';

interface IProps {
  children: any;
  padding?: boolean;
}

export const ThemeProviderWrapper: React.FC<IProps> = ({children}) => {
  return <ThemeProvider theme={Theme}>{children}</ThemeProvider>;
};
