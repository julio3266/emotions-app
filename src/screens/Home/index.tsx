import React from 'react';
import {assignTestId} from '../../utils/qualityAssurance';
import {Header} from '../../Components/Header';
import * as Styled from './styles';

export interface IHomeProps {
  testID?: string;
}

export const Home: React.FC<IHomeProps> = ({testID = 'Home'}) => {
  return (
    <Styled.Container {...assignTestId('StatusBar', `${testID}_statusBar`)}>
      <Header
        {...assignTestId('View', `${testID}_header`)}
        title={'How are you feeling?'}
        isBack={false}
      />
    </Styled.Container>
  );
};
