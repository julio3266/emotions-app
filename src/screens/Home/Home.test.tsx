import React from 'react';
import {render} from '@testing-library/react-native';
import {mockTestID} from '../../utils/qualityAssurance';
import {ThemeProviderWrapper as wrapper} from '../../utils/ThemeProviderWrapper';
import {Home, IHomeProps} from '../Home';

describe('Components/Home', () => {
  let props: IHomeProps = {
    testID: 'Home',
  };
  const {findByTestId, toJSON} = render(<Home {...props} />, {
    wrapper,
  });

  const sut_container = mockTestID('View', props.testID!).testID;
  const sut_header = mockTestID('View', `${props.testID!}_header`).testID;
  it(`should render #${sut_container}`, () => {
    const sut = findByTestId(sut_container);
    expect(sut).toBeTruthy();
  });

  it(`should render #${sut_header}`, () => {
    const sut = findByTestId(sut_header);
    expect(sut).toBeTruthy();
  });

  it(`should render a snapshot`, () => {
    expect(toJSON()).toMatchSnapshot();
  });
});
