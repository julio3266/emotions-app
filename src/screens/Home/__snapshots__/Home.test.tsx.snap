// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Home should render a snapshot 1`] = `null`;
